tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumerOptions.java

```diff
@@ -364,6 +364,10 @@ private static String convertDeprecatedClass(String className) {
                 System.err.println("WARNING: kafka.tools.NoOpMessageFormatter is deprecated and will be removed in the next major release. " +
                         "Please use org.apache.kafka.tools.consumer.NoOpMessageFormatter instead");
                 return NoOpMessageFormatter.class.getName();
+            case "kafka.coordinator.transaction.TransactionLog$TransactionLogMessageFormatter":
+                System.err.println("WARNING: kafka.coordinator.transaction.TransactionLog$TransactionLogMessageFormatter is deprecated and will be removed in the next major release. " +
+                        "Please use org.apache.kafka.tools.consumer.TransactionLogMessageFormatter instead");
+                return className;
```

Comment:
If we can deprecate the old one, how about using `TransactionLogMessageFormatter.class.getName()` here?

Reply:
If user want to use the old `TransactionLog$TransactionLogMessageFormatter`, we should let it work before we remove it, thus show the error message and let user use old formatter, I think is the good way.

Reply:
From JIRA description:
> That is to say, `ConsoleConsumer` can accept the previous package name and then use the (java) implementation to parse and make same output.

If we don't change `className` to `TransactionLogMessageFormatter.class.getName()`, do we use implementation from scala or java?

Reply:
> If we don't change className to TransactionLogMessageFormatter.class.getName(), do we use implementation from scala or java?

that is a good point. The replacement does not generate same output as before. The previous output is not json format, but we want to leverage the protocol json mechanism ...

@m1a2st Could you please update the jira description to avoid misdirecting readers.